output/transform/postgres_relations.go

```diff
@@ -249,22 +251,17 @@ func transformPostgresRelations(s snapshot.FullSnapshot, newState state.Persiste
 			}
 			s.IndexInformations = append(s.IndexInformations, &indexInfo)
 
-			// Statistic
-			if diffedSchemaStatsExist {
```

Comment:
I mentioned this in https://github.com/pganalyze/collector/pull/633#discussion_r1841307149

Reply:
That explains why you removed the exists check of `indexStats, exists := diffedSchemaStats.IndexStats[index.IndexOid]`, but I don't think you mention things about `diffedSchemaStats`. Or if that comment is also covering about `diffedSchemaStats` and you ensured that `diffState.SchemaStats[relation.DatabaseOid]` will never be non-existing, I think we can safely change `diffedSchemaStats, diffedSchemaStatsExist := diffState.SchemaStats[relation.DatabaseOid]` part too.

Reply:
In the linked comment, the goal is to have a 1-1 mapping between the info and statistics arrays, so that `mergePartitionSizes` can rely on that array ordering.

My (possibly incorrect) understanding is that `indexStats := diffedSchemaStats.IndexStats[index.IndexOid]` will return a zero-type struct if the stats are missing. This updated code then uses that zero-type struct to fill in zeros to maintain the 1-1 mapping.

Reply:
Here is a simple go playground, but my point is if `diffedSchemaStats` can be nil or not and we should handle it here too.
https://go.dev/play/p/deAV4_hK9ry

Though, now I understand that you want to ensure that index stats won't be missed due to `diffedSchemaStats` being nil (and you're okay with putting 0 stats values in that case). If so, I think you need to tweak this code a bit more and do that explicitly when `diffedSchemaStats` is nil (otherwise it'll error like above playground).